generator/generator_tests/restriction_collector_test.cpp

```diff
@@ -0,0 +1,170 @@
+  // Adding restrictions and feature ids to restrictionCollector in mixed order.
+  restrictionCollector.AddRestriction(Restriction::Type::No, {1, 2} /* osmIds */);
+  restrictionCollector.AddFeatureId(30 /* featureId */, {3} /* osmIds */);
+  restrictionCollector.AddRestriction(Restriction::Type::No, {2, 3} /* osmIds */);
+  restrictionCollector.AddFeatureId(10 /* featureId */, {1} /* osmIds */);
+  restrictionCollector.AddFeatureId(50 /* featureId */, {5} /* osmIds */);
+  restrictionCollector.AddRestriction(Restriction::Type::Only, {5, 7} /* osmIds */);
+  restrictionCollector.AddFeatureId(70 /* featureId */, {7} /* osmIds */);
+  restrictionCollector.AddFeatureId(20 /* featureId */, {2} /* osmIds */);
+
+  // Composing restriction in feature id terms.
+  restrictionCollector.ComposeRestrictions();
+  restrictionCollector.RemoveInvalidRestrictions();
+
+  // Checking the result.
+  TEST(restrictionCollector.IsValid(), ());
+
+  RestrictionVec const expectedRestrictions = {{Restriction::Type::No, {10, 20}},
+                                               {Restriction::Type::No, {20, 30}},
+                                               {Restriction::Type::Only, {50, 70}}};
+  TEST_EQUAL(restrictionCollector.m_restrictions, expectedRestrictions, ());
+}
+
+UNIT_TEST(RestrictionTest_InvalidCase)
+{
+  RestrictionCollector restrictionCollector("", "");
+  restrictionCollector.AddFeatureId(0 /* featureId */, {0} /* osmIds */);
+  restrictionCollector.AddRestriction(Restriction::Type::No, {0, 1} /* osmIds */);
+  restrictionCollector.AddFeatureId(20 /* featureId */, {2} /* osmIds */);
+
+  restrictionCollector.ComposeRestrictions();
+
+  TEST(!restrictionCollector.IsValid(), ());
+
+  RestrictionVec const expectedRestrictions = {
+      {Restriction::Type::No, {0, Restriction::kInvalidFeatureId}}};
+  TEST_EQUAL(restrictionCollector.m_restrictions, expectedRestrictions, ());
+
+  restrictionCollector.RemoveInvalidRestrictions();
+  TEST(restrictionCollector.m_restrictions.empty(), ());
+  TEST(!restrictionCollector.IsValid(), ());
+}
+
+UNIT_TEST(RestrictionTest_ParseRestrictions)
+{
+  string const kRestrictionName = "restrictions_in_osm_ids.csv";
+  string const kRestrictionPath = my::JoinFoldersToPath(kRestrictionTestDir, kRestrictionName);
+  string const kRestrictionContent = R"(No, 1, 1,
+                                        Only, 0, 2,
+                                        Only, 2, 3,
+                                        No, 38028428, 38028428
+                                        No, 4, 5,)";
+
+  ScopedDir const scopedDir(kRestrictionTestDir);
+  ScopedFile const scopedFile(kRestrictionPath, kRestrictionContent);
+
+  RestrictionCollector restrictionCollector("", "");
+
+  Platform const & platform = Platform();
+
+  TEST(restrictionCollector.ParseRestrictions(
+           my::JoinFoldersToPath(platform.WritableDir(), kRestrictionPath)),
+       ());
+  RestrictionVec expectedRestrictions = {{Restriction::Type::No, 2},
```

Comment:
Because "2" is a size. Pls have a look at Restriction constructor:
```
struct Restriction
{
...
  Restriction(Type type, size_t linkNumber);
  Restriction(Type type, vector<FeatureId> const & links);
...
```